test/integration/short/control-connection-simulator-tests.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const assert = require('assert');
+const simulacron = require('../simulacron');
+const utils = require('../../../lib/utils');
+const types = require('../../../lib/types');
+
+const Client = require('../../../lib/client.js');
+
+describe('ControlConnection', function() {
+  this.timeout(5000);
+
+  before(simulacron.start.bind(simulacron));
+  after(simulacron.stop.bind(simulacron));
+
+  describe.skip("#init", function() {
+    it('should downgrade to protocol v3 with versions 3.0 & 2.1', testWithNodes(['3.0.13', '2.1.17'], 3));
+    it('should downgrade to protocol v3 with versions 2.2 & 2.1', testWithNodes(['2.2.11', '2.1.17'], 3));
+    it('should downgrade to protocol v2 with versions 2.2 & 2.0', testWithNodes(['2.2.11', '2.0.17'], 2));
+    it('should downgrade to protocol v1 with versions 2.2 & 1.2', testWithNodes(['2.2.11', '1.2.19'], 1));
+    it('should downgrade to protocol v2 with versions 2.1 & 2.0', testWithNodes(['2.1.17', '2.0.17'], 2));
+    it('should downgrade to protocol v1 with versions 2.1 & 1.2', testWithNodes(['2.1.17', '1.2.19'], 1));
+    it('should downgrade to protocol v1 with versions 2.0 & 1.2', testWithNodes(['2.0.17', '1.2.19'], 1));
+    // no need to downgrade since both support protocol V4.
+    it('should not downgrade with versions 3.0 & 2.2', testWithNodes(['3.0.13', '3.0.11', '2.2.9'], 4));
+    // can't downgrade because C* 3.0 does not support protocol V2.
+    it('should not downgrade with versions 3.0 & 2.0', testWithNodes(['3.0.13', '2.0.17'], 4));
+    // can't downgrade because C* 3.0 does not support protocol V1.
+    it('should not downgrade with versions 3.0 & 1.2', testWithNodes(['3.0.13', '1.2.19'], 4));
+  });
+});
+
+function testWithNodes(nodeVersions, expectedProtocolVersion, maxVersion) {
+  const nodes = [];
+  for (let i = 0; i < nodeVersions.length; i++) {
+    nodes.push({
+      id: i,
+      cassandra_version: nodeVersions[i]
+    });
+  }
+
+  const clientOptions = {
+    contactPoints: [simulacron.startingIp],
+  };
+
+  if (maxVersion) {
+    clientOptions.protocolOptions = { maxVersion: maxVersion };
+  }
+
+  const usedMaxVersion = maxVersion ? maxVersion : types.protocolVersion.maxSupported;
+
+  return function (done) {
+    const client = new Client(clientOptions);
+
+    function cleanUp(err) {
+      client.shutdown(() => cluster.unregister(function() {
+        assert.ifError(err);
+        done();
+      }));
+    }
+
+    const cluster = new simulacron.SimulacronCluster();
+    utils.series([
+      function register(next) {
+        cluster.registerWithBody({
+          data_centers: [
+            {
+              id: 0,
+              name: "dc0",
+              nodes: nodes
+            }
+          ]
+        }, next);
+      },
+      function connect(next) {
+        client.connect(function (err, result) {
+          // TODO: Check for downgrade.
+          if (expectedProtocolVersion < 3) {
```

Comment:
Because simulacron doesn't support protocol v1 and v2 (as native-protocol library it depends on doesn't support it), we expect an error in this case.  I'll also add a query log message in simulacron when there is an attempt with a bad protocol version so we can validate it.    I'll probaly also add an integration test with a 2 node cluster with C* 2.1 and C* 2.0 to test that downgrade case.

Reply:
I added a long integration test for this, takes about ~80seconds (as after bootstrapping a node it takes 30 seconds to join, waits for gossip to settle, etc.), but looks to be working.